components/ConvertSteps/ManageStep.js

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useCallback, useState } from 'react'
+import PropTypes from 'prop-types'
+import Step from './Step'
+
+function ManageStep({
```

Comment:
I think this works well for now, but as we generalize this set of stepper components and try to bring it to the client/dashboard in a re-usable way, I think we may very quickly want to lift this "management" up to the `ManageConversion` or `Stepper`'s parent.

Now that we have the tuple API, might eventually go towards something like this, where all the logic is lifted out of the view components:

```
---------------------
|      Parent       |
|      (logic)      |
|                   |
|  ---------------  |
|  |  Stepper    |  |
|  |  (agnostic) |  |
|  |             |  |
|  |  ---------  |  |
|  |  | Steps |  |  |
|  |  ---------  |  |
|  ---------------  |
---------------------
```

Reply:
I agree in concept. My original goals with having individual step status managed in the children like this was to incapsulate that behaviour rather than trying to pass it down via a more complex state object or similar.

I'd be interested to discuss what this composition could look like, but yes, let's cross that bridge when we come to it.